File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace T362981MVC
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function
		Public Function GridViewPartial() As ActionResult
		   Dim model = Enumerable.Range(0, 10).Select(Function(i) New With {Key .ProductID = i, Key .ProductName1 = "ProductName " & i, Key .ProductName2 = "ProductName " & i, Key .ProductName3 = "ProductName " & i, Key .ProductName4 = "ProductName " & i, Key .ProductName5 = "ProductName " & i})
			Return PartialView(model)
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace T362981MVC {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult GridViewPartial() {
           var model =  Enumerable.Range(0, 10).Select(i => new
            {
                ProductID = i,
                ProductName1 = "ProductName " + i,
                ProductName2 = "ProductName " + i,
                ProductName3 = "ProductName " + i,
                ProductName4 = "ProductName " + i,
                ProductName5 = "ProductName " + i
            });
            return PartialView(model);
        }
    }
}